posydon/popsyn/io.py

```diff
@@ -487,13 +487,18 @@ def binarypop_kwargs_from_ini(path, verbose=False):
             for key, val in parser[section].items():
                 pop_kwargs[key] = ast.literal_eval(val)
 
-            # hard code for running with MPI, only try to import if use_MPI
-            if pop_kwargs['use_MPI']:
-                from mpi4py import MPI
-                pop_kwargs['comm'] = MPI.COMM_WORLD
+            # Check if we are running as a job array
+            JOB_ID = os.getenv('SLURM_ARRAY_JOB_ID')
+            if JOB_ID is not None:
+                pop_kwargs['JOB_ID'] = np.int64(os.environ['SLURM_ARRAY_JOB_ID'])
+                # account for job array not starting at 0
+                min_rank = np.int64(os.environ['SLURM_ARRAY_TASK_MIN'])
+                pop_kwargs['RANK'] = np.int64(os.environ['SLURM_ARRAY_TASK_ID'])-min_rank
+                pop_kwargs['size'] = np.int64(os.environ['SLURM_ARRAY_TASK_COUNT'])
```

Comment:
Are we sure, that 'SLURM_ARRAY_TASK_MIN', 'SLURM_ARRAY_TASK_ID' and 'SLURM_ARRAY_TASK_COUNT' always exist if 'SLURM_ARRAY_JOB_ID' does?

Reply:
Yes. These are always spawned at the same time.